doc/designs/quic-design/quic-record-layer.md

```diff
@@ -0,0 +1,101 @@
+Design Problem: Record Layer
+============================
+
+* BIO filter
+    * Later exposed via provider
+    * Ctrls
+* RECORD METHOD structure
+    * Later exposed via provider
+    * Could be individual callback mechanism
+    * PARAMS + arguments
+
+The record layer transmission is a simple layer which takes data from
+upper levels, adds protection and (possibly) record headers and passes it
+to the next layer for sending. Receiving packets is the opposite
+operation: data is received from the network layer, protection is removed
+and plaintext data is passed to the upper layers. The record layer API
+needs to be universal - it needs to work for TLS, DTLS, KTLS and QUIC.
+
+Definition of _single-copy_: Given there needs to happen an encryption
+or decryption to protect or unprotect the data send or received, a copy
+of the data (in the transformed form) is always created. Thus the
+usual term of _zero-copy_ does not make sense here and we use the term
+_single-copy_.
```

Comment:
This still seems to lack things that our design policy says we need as input.

Reply:
Could you please be more concrete?

Reply:
From the document:
Inputs:

Prior to commencing design work the OTC shall identify:
- The business requirements for the problem to be solved (obtained from the OMC)
- The scope of the specific problem area to be addressed
- Any additional design constraints that the OTC wishes to impose
- Any other constraints that may apply (e.g. timescales)
- The individual(s) that will work on developing the design (referred to as the "design team" in this document)

It's not clear at all which problem it's trying to address.